toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/panels/AbstractPanel.java

```diff
@@ -0,0 +1,177 @@
+package io.toast.tk.agent.ui.panels;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Image;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import javax.swing.*;
+
+import io.toast.tk.agent.ui.utils.PanelHelper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public abstract class AbstractPanel extends JPanel {
+
+
+	private static final long serialVersionUID = 1L;
+
+	private static final Logger LOG = LogManager.getLogger(AbstractPanel.class);
+	private final Properties properties;
+	private static Dimension dim;
+
+	protected String strkey;
+	protected JTextField textField;
+	protected JPanel iconPanel;
+	protected JLabel errorLabel;
+
+	protected JLabel iconValid, iconNotValid;
+
+	private Image toastLogo;
+	private Image notvalidImage;
+	private Image validImage;
+
+	protected final String errorMessageSelectUrl = "The URL does not anwser.";
+	private final String errorMessageApiKey = "The Api Key have to match with the WebApp";
+	
+	public AbstractPanel(Properties properties, String strkey) throws IOException {
+		this.properties = properties;
+		this.strkey = strkey;
+	}
+	
+	protected void setBasicProperties(String strkey) throws IOException {
+		this.strkey = strkey;
+		
+		this.notvalidImage = PanelHelper.createImage(this,"picto-non-valide.png");
+		this.validImage = PanelHelper.createImage(this,"picto-valide.png");
+		this.toastLogo = PanelHelper.createImage(this,"ToastLogo_24.png");
+				
+		this.buildContentPanel();
+		
+		if(dim == null) {
+			dim = getPreferredSize();
+		}
+		else {
+			setPreferredSize(dim);
+		}		
+	}
+	
+	public String getTextValue() {
+		return this.textField.getText();
+	}
+
+	private JTextField createBasicTextPanel() {
+		JTextField textField = new JTextField(render(properties.getProperty(strkey)));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "textField" which hides the field declared at line 31. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)